static/js/recent_topics_ui.js

```diff
@@ -810,9 +810,11 @@ function is_focus_at_last_table_row() {
 }
 
 function has_unread(row) {
+    // Based on assumption that there cannot be a read message below an
+    // unread message.
     const last_msg_id = topics_widget.get_current_list()[row].last_msg_id;
     const last_msg = message_store.get(last_msg_id);
-    return unread.num_unread_for_topic(last_msg.stream_id, last_msg.topic) > 0;
+    return last_msg.unread;
```

Comment:
This is not a safe assumption; for example, messages you yourself sent won't be marked as unread. We need to fetch the number of unreads anyway in order to display the count, so I don't see much benefit in trying to skip that.